situp-plugins/elasticsearch/src/test/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSinkIT.java

```diff
@@ -276,4 +328,111 @@ private boolean deleteDirectory(final File directoryToBeDeleted) {
     }
     return directoryToBeDeleted.delete();
   }
+
+  public static boolean isODFE() {
+    final boolean isODFE = Optional.ofNullable(System.getProperty("odfe"))
+            .map("true"::equalsIgnoreCase).orElse(false);
+    if (isODFE) {
+      // currently only external cluster is supported for security enabled testing
+      if (!Optional.ofNullable(System.getProperty("tests.rest.cluster")).isPresent()) {
+        throw new RuntimeException("cluster url should be provided for security enabled ODFE testing");
+      }
+    }
+
+    return isODFE;
+  }
+
+  @Override
+  protected String getProtocol() {
+    return isODFE() ? "https" : "http";
+  }
+
+  @Override
+  protected RestClient buildClient(final Settings settings, final HttpHost[] hosts) throws IOException {
+    final RestClientBuilder builder = RestClient.builder(hosts);
+    if (isODFE()) {
+      configureHttpsClient(builder, settings);
+    } else {
+      configureClient(builder, settings);
+    }
+
+    builder.setStrictDeprecationMode(true);
+    return builder.build();
+  }
+
+  @SuppressWarnings("unchecked")
+  @After
+  protected void wipeAllODFEIndices() throws IOException {
+    final Response response = client().performRequest(new Request("GET", "/_cat/indices?format=json&expand_wildcards=all"));
+    final XContentType xContentType = XContentType.fromMediaTypeOrFormat(response.getEntity().getContentType().getValue());
+    try (
+            XContentParser parser = xContentType
+                    .xContent()
+                    .createParser(
+                            NamedXContentRegistry.EMPTY,
+                            DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                            response.getEntity().getContent()
+                    )
+    ) {
+      final XContentParser.Token token = parser.nextToken();
+      List<Map<String, Object>> parserList = null;
+      if (token == XContentParser.Token.START_ARRAY) {
+        parserList = parser.listOrderedMap().stream().map(obj -> (Map<String, Object>) obj).collect(Collectors.toList());
+      } else {
+        parserList = Collections.singletonList(parser.mapOrdered());
+      }
+
+      for (final Map<String, Object> index : parserList) {
+        final String indexName = (String) index.get("index");
+        if (indexName != null && !".opendistro_security".equals(indexName)) {
+          client().performRequest(new Request("DELETE", "/" + indexName));
+        }
+      }
+    }
+  }
+
+  protected static void configureHttpsClient(final RestClientBuilder builder, final Settings settings) throws IOException {
```

Comment:
Can we get away with calling configureClient and then just adding the HTTPs stuff?